### PR TITLE
Save sort order of constraints summary for duration of session

### PR DESCRIPTION
**Pull request by toddburnside**
Not yet being saved to the user preferences service.

### CODE REVIEW

File: explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala [hunk 1]
```diff
@@ -73,10 +76,20 @@ object ConstraintsSummaryTable {
...
+  private val columnClasses: Map[String, Css]                                             = Map(
     "edit" -> (ExploreStyles.Sticky |+| ExploreStyles.ConstraintsSummaryEdit)
   )
 
+  private def toSortingRules(tuples: List[(String, Boolean)])                             = tuples.map { case (id, b) =>
+    SortingRule[ConstraintGroup](id).setDesc(b)
+  }.toJSArray
+
+  private def fromTableState(state: TableState[ConstraintGroup]): List[(String, Boolean)] = state
+    .asInstanceOf[ConstraintsTable.StateType]
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala:88]: I would expect things to work if we define `state: ConstraintsTable.StateType` and avoid the cast.

**Review by rpiaggio — APPROVED** [review]
LGTM.

File: explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala (hunk 1 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala:88]: Except that `tableInstance.state` returns a `TableState` (no sorting state mixed in). I don't know of another way to obtain the state, but I may be missing something.

Comment (rpiaggio) [reply explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala:88]: Ah, right `TableInstance` facade is generated by ScalablyTyped and is not type-parametrized beyond `D`. Sorry, got mixed up.

**Comment by cquiroz** [comment]
Should we store this in the persistent user preferences?

**Comment by toddburnside** [comment]
> Should we store this in the persistent user preferences?

That's the next step.

**Pull request merged**